src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/DataAnnotationsMetadataProvider.cs

```diff
@@ -22,9 +23,13 @@ public class DataAnnotationsMetadataProvider :
         IValidationMetadataProvider
     {
         private readonly IStringLocalizerFactory _stringLocalizerFactory;
+        private readonly IOptions<MvcDataAnnotationsLocalizationOptions> _options;
 
-        public DataAnnotationsMetadataProvider(IStringLocalizerFactory stringLocalizerFactory)
+        public DataAnnotationsMetadataProvider(
+            IOptions<MvcDataAnnotationsLocalizationOptions> options,
+            IStringLocalizerFactory stringLocalizerFactory)
         {
+            _options = options;
```

Comment:
We `null`-check `IOptions<TOptions>` parameters everywhere else. Should also `throw` an `ArgumentNullException` here.